src/Deployment/FtpServer.php

```diff
@@ -278,10 +277,25 @@ private function protect(callable $func, $args = [])
 				$message = $m[1];
 			}
 			throw new FtpException($message);
-		});
-		$res = call_user_func_array($func, $args);
+		};
+
+		set_error_handler($ftpExceptionErrorHandler);
```

Comment:
This must be compatible with PHP 5.4 (composer.json, .travis.yml) so the code is ugly.

For PHP >= 5.5 would be something like this (shorter, cleaner):

```php
	set_error_handler($ftpExceptionErrorHandler);

	try {
		return call_user_func_array($func, $args);
	} finally {
		restore_error_handler();
	}
```